pkg/cmd/cli/cmd/process.go

```diff
@@ -178,8 +178,7 @@ func RunProcess(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []
 			if len(templateName) > 0 {
 				sourceName = namespace + "/" + templateName
 			}
-			fmt.Fprintf(cmd.Out(), "unable to parse %q, not a valid Template but %s\n", sourceName, reflect.TypeOf(infos[i].Object))
-			continue
+			return fmt.Errorf("unable to parse %q, not a valid Template but %s\n", sourceName, reflect.TypeOf(infos[i].Object))
```

Comment:
The argument against allowing more than one template to be passed in as a list was that it makes validation of user intent pretty hard -- you can't really tell if they are missing parameters or have superfluous ones, etc, without reading all of the templates in at once.